chain/client/src/info.rs

```diff
@@ -268,12 +272,14 @@ impl InfoHelper {
             paint(ansi_term::Colour::Blue, machine_info_log),
         );
         if catchup_status_log != "" {
-            info!(target:"stats", "Catchups\n{}", catchup_status_log);
+            info!(target: "stats", "Catchups\n{}", catchup_status_log);
         }
         if let Some(statistics) = statistics {
             rocksdb_metrics::export_stats_as_metrics(statistics);
         }
-
+        if let Some(config_updater) = &config_updater {
+            config_updater.log_error();
```

Comment:
I am not sure about passing the error all the way from neard to client and to here just for logging. Without this, the logic for ConfigUpdater is much simpler as the struct does not contain any state. I see two other options: 
- Simply let the node panic in neard after loading the new config if the config is updated to an invalid setting. This may be a little harsh, but I think it is also reasonable. When restarting the node, it panics if the config is invalid. So I think it is ok to terminate the node if the updated config is invalid. 
- Log the error into a separate file such as `config_log.txt`. This way, the error message will not be lost in the other log messages, so you don't have store the message and log it every time with the `info()`.

Reply:
We have a tradeoff between code complexity and making it more user-friendly. Or rather, more difficult to shoot yourself in the foot and not notice it.

I lean strongly on the side of user-friendliness.
There is also a metric exported about the config being invalid.

> let the node panic

That is user-anti-friendly. I can imagine updating the config during an incident, and accidentally killing a very valuable node. Don't want that.
And if a validator accidentally kills a node and misses blocks, that is not good either.

> `config_log.txt`

That is non-trivial to use properly :thinking:
I would like to use it like

```
killall -SIGHUP neard ; cat ~/.near/config_log.txt || echo 'Config valid'
```

But it may take longer than a moment to apply that config change, parse the config and validate it.
To be sure it's applied a delay is needed.

```
killall -SIGHUP neard ; sleep 10; cat ~/.near/config_log.txt || echo 'Config valid'
```

But I also cannot guarantee that `10` seconds is a sufficient delay.
And anyway, if I can remember to check `config_log.txt`, I can remember to check the Prometheus metric.